src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private double radius;
+
+    public void printInfo() {
+        System.out.println("Figure: " + getName()
+                + ", " + "area: " + calculateArea()
+                + ", " + "radius: " + getRadius()
+                + ", " + "color: " + getColor());
+
+    }
+
+    @Override
+    public double calculateArea() {
+        return Math.PI * (radius * radius);
+    }
+
+    public void generateRandomParameters() {
+        radius = random.nextDouble() * 20;
+        name = "Circle";
+        color = colorSupplier.getRandomColor();
+    }
```

Comment:
Circle class should represent "Circle" and actions with circle, e.g. calculate area. 
It shouldn't generate itself random parameter, for this reason we have FigureSupplier.

Refactor entire code according this comment. 

```suggestion
```